scripts/integration-test.sh

```diff
@@ -1,6 +1,531 @@
+# if the words ERROR or CRIT are in the log file, send a message about it to
+# stdout, cat the log file, then pass the error along with a message to the
+# 'fatal' function
+#------------------------------------------------------------------------------
+function cat_log_on_log_error() {
+  log_base_name=$1
+  for err_type in ERROR CRIT
+  do
+    grep $err_type $log_base_name.log > /dev/null
+    err_found=$?
+    if [ $err_found = 0 ]
+    then
+      echo "$log_base_name.log contanins $err_type"
+      cat $log_base_name.log
+      fatal $err_found "$log_base_name $err_type terminated this test"
+    fi
+  done
+}
+
+#------------------------------------------------------------------------------
+# send in an error code and the base name of a log file.
+# offer these to the error code test function 'cat_log_on_log_error' and the
+# the log file test function 'cat_log_on_error'
+#------------------------------------------------------------------------------
+function check_for_logged_fatal_errors() {
+  error_code=$1
+  log_base_name=$2
+  cat_log_on_log_error $log_base_name
+  cat_log_on_error $error_code $log_base_name "much to your displeasure"
+}
+
+#------------------------------------------------------------------------------
+# iterate through the names of the RabbitMQ queues clearing them out using the
+# 'test_rabbitmq.py' Socorro app. Send the result to the error reporting
+# function 'check_for_logged_fatal_errors'
+#------------------------------------------------------------------------------
+function cleanup_rabbitmq() {
+  for queue_name in $rmq_normal_queue_name $rmq_priority_queue_name $rmq_reprocessing_queue_name
+  do
+    echo "INFO: Purging rabbitmq queue" $queue_name
+    socorro socorro.external.rabbitmq.test_rabbitmq.TestRabbitMQApp \
+        --test_rabbitmq.purge=$queue_name \
+        --test_rabbitmq.rabbitmq_host=$rmq_host \
+        --test_rabbitmq.rabbitmq_user=$rmq_user \
+        --test_rabbitmq.rabbitmq_password=$rmq_password \
+        --test_rabbitmq.rabbitmq_vhost=$rmq_virtual_host \
+        > $queue_name.log 2>&1
+    check_for_logged_fatal_errors $? $queue_name
+  done
+}
+
+#------------------------------------------------------------------------------
+# empty the rabbit queues, clean the local filesystem, kill any apps that this
+# shell script may have started
+#------------------------------------------------------------------------------
+function cleanup() {
+  cleanup_rabbitmq
+
+  echo "INFO: cleaning up crash storage directories"
+  rm -rf ./primaryCrashStore/ ./processedCrashStore/
+  rm -rf ./crashes/
+  rm -rf ./submissions
+
+  echo "INFO: Terminating background jobs"
+  echo "  any kill usage errors below may be ignored"
+
+  for p in collector processor middleware
+  do
+    # destroy any running processes started by this shell
+    kill $(jobs -p) > /dev/null 2>&1
+    # destroy anything trying to write to the log files too
+    fuser -k ${p}.log > /dev/null 2>&1
```

Comment:
too much information, I wanted to be able to read through the clutter.  btw, this is a direct quote of the existing code.